settings/src/components/email-address.js

```diff
@@ -0,0 +1,94 @@
+
+/**
+ * WordPress dependencies
+ */
+import { Button, TextControl, Notice, Spinner } from '@wordpress/components';
+import { useState, useEffect } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as coreDataStore, useEntityRecord } from '@wordpress/core-data';
+
+/**
+ * Render the Email setting.
+ */
+export default function EmailAddress( { userId } ) {
+	const { record, edit, save, editedRecord, hasEdits } = useEntityRecord( 'root', 'user', userId );
+
+	// useEntityRecord() doesn't expose isSaving. Huh.
+	const isSaving = useSelect( ( select ) => select( coreDataStore ).isSavingEntityRecord( 'root', 'user', userId ) );
+
+	const [ emailError, setEmailError ] = useState( '' );
+	const [ justChangedEmail, setJustChangedEmail ] = useState( false );
+
+	const handleEmailChange = ( email ) => edit( { email } );
+
+	const handleSave = async () => {
+		try {
+			await save();
+
+			setJustChangedEmail( true );
+		} catch( error ) {
+			// TODO: Change these texts
+			// error.code: rest_user_invalid_email (can't use that one), rest_invalid_param (invalid email format)
+			// TODO: The red paragraph inserted inline feels a bit hacky.
+			setEmailError( error.message );
+		}
+	};
+
+	const handleDiscard = async () => {
+		try {
+			await edit( { pending_email: '' } );
+			await save();
+		} catch( error ) {
+			alert( error.message );
+		}
+	};
```

Comment:
Should all of these use `useCallback` to avoid re-defining them during each render?

I think that's important in some cases, but it might just be a micro-optimization here.

Reply:
I haven't learnt `useCallback` yet :)

I think you're right, it's likely micro-optimization here, but there's no reason not to use best practices for the smallest of things.